cwf/cloud/go/services/cwf/obsidian/handlers/handlers.go

```diff
@@ -317,6 +326,131 @@ func getHealthStatusHandler(c echo.Context) error {
 	return c.JSON(http.StatusOK, healthState)
 }
 
+func listHAPairsHandler(c echo.Context) error {
+	nid, nerr := obsidian.GetNetworkId(c)
+	if nerr != nil {
+		return nerr
+	}
+	ids, err := configurator.ListEntityKeys(nid, cwf.CwfHAPairType)
+	if err != nil {
+		return obsidian.HttpError(err, http.StatusInternalServerError)
+	}
+	haPairTKs := make([]storage.TypeAndKey, 0, len(ids))
+	for _, id := range ids {
+		haPairTKs = append(haPairTKs, storage.TypeAndKey{Type: cwf.CwfHAPairType, Key: id})
+	}
+	haPairEnts, _, err := configurator.LoadEntities(
+		nid,
+		swag.String(cwf.CwfHAPairType),
+		nil,
+		nil,
+		haPairTKs,
+		configurator.FullEntityLoadCriteria(),
+	)
+	if err != nil {
+		return obsidian.HttpError(err, http.StatusInternalServerError)
+	}
+	haPairEntsByTK := haPairEnts.MakeByTK()
```

Comment:
This looks like an unnecessary conversion here, you can just iterate through the slice